### PR TITLE
feat(sql console rule): sql console adapt sql type of call ,comment on and set session

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
sql console adapt sql type extension.
The following syntax types are supported:
1. COMMENT ON COLUMN table_name.column_name IS 'XXX';
2. CALL procedureName();
3. SET SESSION time_zone = '+00:00';
4. ALTER SESSION SET time_zone = '+00:00';

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/model/SqlExecuteResult.java [hunk 1]
```diff
@@ -216,7 +216,8 @@ private boolean checkContainsExternalTable(@NonNull ConnectionSession connection
         DialectType dialectType = connectionSession.getDialectType();
         if (dialectType == DialectType.OB_MYSQL || dialectType == DialectType.OB_ORACLE) {
             String obVersion = ConnectionSessionUtil.getVersion(connectionSession);
-            if (VersionUtils.isGreaterThanOrEqualsTo(obVersion, MIN_OB_VERSION_FOR_EXTERNAL_TABLE)) {
+            if (VersionUtils.isGreaterThanOrEqualsTo(obVersion, MIN_OB_VERSION_FOR_EXTERNAL_TABLE)
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/model/SqlExecuteResult.java:219]: What's this change for?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java [hunk 2]
```diff
@@ -55,6 +55,7 @@ public static GeneralSqlType getGeneralSqlType(BasicResult result) {
             case DROP:
             case TRUNCATE:
             case ALTER:
+            case COMMENT:
```
Comment (MarkPotato777) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java:58]: What about CALL and SET_SESSION?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java [hunk 3]
```diff
@@ -38,6 +38,9 @@ public enum SqlType {
     ALTER,
     TRUNCATE,
     CREATE,
+    CALL,
+    COMMENT,
+    SET_SESSION,
```
Comment (MarkPotato777) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java:43]: Is `SET_SESSION` a subset of `SET` statement? If yes, we may not add a single SET_SESSION SQL Type, it's kind of wired.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/model/SqlExecuteResult.java (hunk 1 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/model/SqlExecuteResult.java:219]: Used to fix npe about external table. Commit it together in this pr for convenience

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java (hunk 2 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java:58]: The call statement is used to invoke procedures and functions

Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java:58]: The `SET SESSION` statement belongs to the syntax of mysql. For example, SET SESSION time_zone = '+00:00';

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java (hunk 3 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java:43]: ok

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java (hunk 2 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/ParserUtil.java:58]: call and set_session belong to unknow

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java (hunk 3 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java:43]: set session should be separated from set. set session is less dangerous than set globle

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java [hunk 4]
```diff
@@ -159,4 +159,11 @@ public void enterSp_decl(Sp_declContext ctx) {
         varibale.setVarType(type);
         this.varibaleList.add(varibale);
     }
+
+    @Override
+    public void enterCall_sp_stmt(PLParser.Call_sp_stmtContext ctx) {
+        this.sqlType = SqlType.CALL;
+        this.dbObjectType = DBObjectType.OTHERS;
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java:166]: why others? 
setting `PROCEDURE` may be better?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModePLParserListener.java [hunk 5]
```diff
@@ -555,6 +555,12 @@ public void enterAlter_package_stmt(Alter_package_stmtContext ctx) {
         }
     }
 
+    @Override
+    public void enterCall_spec(PLParser.Call_specContext ctx) {
+        this.sqlType = sqlType.CALL;
+        this.dbObjectType = DBObjectType.OTHERS;
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModePLParserListener.java:561]: same as mysql listener

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java [hunk 6]
```diff
@@ -418,6 +418,26 @@ public void enterRollback_statement(PlSqlParser.Rollback_statementContext ctx) {
         doRecord(SqlType.ROLLBACK, DBObjectType.OTHERS, null);
     }
 
+    @Override
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:421]: what about set_session statement?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java [hunk 7]
```diff
@@ -418,6 +418,26 @@ public void enterRollback_statement(PlSqlParser.Rollback_statementContext ctx) {
...
+        doRecord(SqlType.COMMENT, DBObjectType.TABLE, null);
+    }
+
+    @Override
+    public void enterComment_on_materialized(PlSqlParser.Comment_on_materializedContext ctx) {
+        doRecord(SqlType.COMMENT, DBObjectType.OTHERS, null);
+    }
+
+    @Override
+    public void enterCall_spec(PlSqlParser.Call_specContext ctx) {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:437]: this ast node is wrong, I think what you wants to handle is `procedure_call`?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 8]
```diff
@@ -700,4 +700,62 @@ public void enterRollback_stmt(Rollback_stmtContext ctx) {
...
+        }
+        String sqlType = child != null ? child.getText() : "";
+        switch (sqlType) {
+            case "TABLE":
+                this.dbObjectType = DBObjectType.TABLE;
+                break;
+            case "COLUMN":
+                this.dbObjectType = DBObjectType.COLUMN;
+                break;
+            case "VIEW":
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:719]:
```
set_comment_stmt
    : COMMENT ON TABLE normal_relation_factor IS STRING_VALUE
    | COMMENT ON COLUMN column_definition_ref IS STRING_VALUE
    ;
```

we just support set comment on table and column~ why should we support so many objects?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 9]
```diff
@@ -700,4 +700,62 @@ public void enterRollback_stmt(Rollback_stmtContext ctx) {
...
+            default:
+                this.dbObjectType = DBObjectType.OTHERS;
+                break;
+        }
+    }
+
+    @Override
+    public void enterCall_stmt(OBParser.Call_stmtContext ctx) {
+        setSqlType(SqlType.CALL);
+        this.dbObjectType = DBObjectType.OTHERS;
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:758]: same as above

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 10]
```diff
@@ -700,4 +700,62 @@ public void enterRollback_stmt(Rollback_stmtContext ctx) {
         this.dbObjectType = DBObjectType.OTHERS;
     }
 
+    @Override
+    public void enterSet_comment_stmt(OBParser.Set_comment_stmtContext ctx) {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:704]: what about set_session statement?

Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:704]: oracle does not support the set session statement

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 8 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:719]: ok

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java (hunk 4 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java:166]: the statement like `call functionName()` also enters this method, so use OTHERS is better

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModePLParserListener.java (hunk 5 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModePLParserListener.java:561]: the statement like call functionName() also enters this method, so use OTHERS is better

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java (hunk 6 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:421]: oracle dont support set_session

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 9 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:758]: the statement like call functionName() also enters this method, so use OTHERS is better

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java (hunk 7 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:437]: removed enterCall_spec

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 8 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:719]: comment on table xxx is "XXX" dont enter `set_comment_stmt` method

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java (hunk 4 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/MysqlModePLParserListener.java]: use PROCEDURE

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java (hunk 6 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:421]: have added

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 10 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:705]: have added

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java [hunk 11]
```diff
@@ -19,25 +19,39 @@
...
+    OTHERS(null),
+    UNKNOWN(null);
+
+    private final SqlType fatherType;
+
+    SqlType(SqlType fatherType) {
+        this.fatherType = fatherType;
+    }
+
+    public SqlType getFatherType() {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java:54]: parentType may be better~

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java [hunk 12]
```diff
@@ -418,6 +417,28 @@ public void enterRollback_statement(PlSqlParser.Rollback_statementContext ctx) {
         doRecord(SqlType.ROLLBACK, DBObjectType.OTHERS, null);
     }
 
+    @Override
+    public void enterComment_on_column(PlSqlParser.Comment_on_columnContext ctx) {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:421]: leak of alter_session

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 13]
```diff
@@ -554,7 +555,7 @@ public void enterAlter_index_stmt(Alter_index_stmtContext ctx) {
     @Override
     public void enterAlter_system_stmt(Alter_system_stmtContext ctx) {
         setSqlType(SqlType.ALTER);
-        this.dbObjectType = DBObjectType.OTHERS;
+        this.dbObjectType = DBObjectType.GLOBAL_VARIABLE;
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:558]: why global variable?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 14]
```diff
@@ -601,8 +602,8 @@ public void enterAlter_sequence_stmt(Alter_sequence_stmtContext ctx) {
 
     @Override
     public void enterAlter_session_stmt(Alter_session_stmtContext ctx) {
-        setSqlType(SqlType.ALTER);
-        this.dbObjectType = DBObjectType.OTHERS;
+        setSqlType(SqlType.ALTER_SESSION);
+        this.dbObjectType = DBObjectType.SESSION_VARIABLE;
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:606]: why session variables?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java [hunk 15]
```diff
@@ -700,4 +701,33 @@ public void enterRollback_stmt(Rollback_stmtContext ctx) {
         this.dbObjectType = DBObjectType.OTHERS;
     }
 
+    @Override
+    public void enterSet_comment_stmt(OBParser.Set_comment_stmtContext ctx) {
+        // comment on xxx
+        setSqlType(SqlType.COMMENT_ON);
+        ParseTree child = null;
+        if (ctx.getChildCount() >= 3) {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:709]: magic number

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/interceptor/SqlConsoleInterceptor.java [hunk 16]
```diff
@@ -194,6 +194,15 @@ private boolean handle(@NonNull SqlAsyncExecuteReq request, @NonNull SqlAsyncExe
         return allowExecute.get();
     }
 
+    private boolean isViolatedRule(Optional<List<String>> allowSqlTypesOpt, BasicResult parseResult) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/interceptor/SqlConsoleInterceptor.java:197]: `Optional<List<String>>` ??
use `List<String>` instead

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java (hunk 11 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/constant/SqlType.java:54]: ok

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java (hunk 12 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeParserListener.java:421]: ok

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 13 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:558]: As before, use others

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 14 above)
Comment (zijiacj) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:606]: As before, use others

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/interceptor/SqlConsoleInterceptor.java (hunk 16 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/interceptor/SqlConsoleInterceptor.java:197]: ok

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java (hunk 15 above)
Comment (yhilmare) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/parser/listener/OracleModeSqlParserListener.java:709]: this way is fine

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by PeachThinking — APPROVED** [review]
lgtm

**Pull request merged**